### PR TITLE
Fix multi-byte UFT-8 in escape('html_attr')

### PR DESCRIPTION
**Pull request by nicolas-grekas**
Alternative to #2734

### CODE REVIEW

File: lib/Twig/Extension/Core.php [hunk 1]
```diff
@@ -1046,21 +1037,9 @@ function twig_escape_filter(Twig_Environment $env, $string, $strategy = 'html',
...
-                static $entityMap = array(
-                    34 => 'quot', /* quotation mark */
-                    38 => 'amp',  /* ampersand */
-                    60 => 'lt',   /* less-than sign */
-                    62 => 'gt',   /* greater-than sign */
-                );
 
                 $chr = $matches[0];
-                $ord = ord($chr);
+                $ord = \ord($chr);
```
Comment (fabpot) [inline lib/Twig/Extension/Core.php]: no need as we are not namespaced.

File: lib/Twig/Extension/Core.php [hunk 2]
```diff
@@ -1074,23 +1053,32 @@ function twig_escape_filter(Twig_Environment $env, $string, $strategy = 'html',
                  * Check if the current character to escape has a name entity we should
                  * replace it with while grabbing the hex value of the character.
                  */
-                if (1 == strlen($chr)) {
-                    $hex = strtoupper(substr('00'.bin2hex($chr), -2));
-                } else {
-                    $chr = twig_convert_encoding($chr, 'UTF-16BE', 'UTF-8');
-                    $hex = strtoupper(substr('0000'.bin2hex($chr), -4));
-                }
+                if (1 === \strlen($chr)) {
```
Comment (fabpot) [inline lib/Twig/Extension/Core.php]: same here

**Review by fabpot — CHANGES_REQUESTED** [review]

File: lib/Twig/Extension/Core.php (hunk 1 above)
Comment (nicolas-grekas) [reply lib/Twig/Extension/Core.php]: fixed

**Comment by fabpot** [comment]
Thank you @nicolas-grekas.

**Pull request merged**